### ISSUE TITLE
existing model compatibility

### ISSUE THREAD

**Issue by twmht** [yjxiong/caffe#191]
Hi,

How can I transfer the existing caffe model to this optimized version. Since you changed a lot, for example, batch normalization layer, I am not sure how to convert an existing caffe model from caffe model zoo(https://github.com/BVLC/caffe/wiki/Model-Zoo).

thank you!

**Comment by yjxiong** [comment]
Add any layer you want in the same manner as adding new layers. Then you can use the model you want.

**Comment by twmht** [comment]
@yjxiong 

What I mean is that I can not load the caffe model from model zoo.

for example, here is resnet-10 (https://github.com/cvjena/cnn-models/blob/master/ResNet_preact/ResNet10_cvgj/deploy.prototxt)

when using this optimized version of caffe, I will get the following errors

```
F1212 23:02:25.500097 28028 layer_factory.hpp:77] Check failed: registry.count(type) == 1 (0 vs. 1) Unknown layer type: BatchNorm (known types: AbsVal, Accuracy, ArgMax, BN, BNLL, BatchReduction, Bias, Concat, ContrastiveLoss, Convolution, Data, Deconvolution, Dropout, DummyData, Eltwise, EuclideanLoss, Exp, Filter, Flatten, Gather, HDF5Data, HDF5Output, HingeLoss, Im2col, ImageData, InfogainLoss, InnerProduct, LRN, Log, MVN, MemoryData, MultinomialLogisticLoss, Normalize, PReLU, Pooling, Power, Python, ROIPooling, ReLU, Reduction, Reshape, SPP, Scale, Scatter, SegData, Sigmoid, SigmoidCrossEntropyLoss, Silence, Slice, SmoothL1Loss, Softmax, SoftmaxWithLoss, Split, TanH, Threshold, VideoData, WindowData)
```

Since you have fused batchNorm with scale/shift, I have no idea how to load such model.

Thank you.

**Comment by twmht** [comment]
@yjxiong 

since it's related to #148 , this issue can be closed.
